pkg/apis/migration/v1alpha1/verify.go

```diff
@@ -0,0 +1,248 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Types
+const (
+	UnhealthySourceNamespaces      = "UnhealthySourceNamespaces"
+	UnhealthyDestinationNamespaces = "UnhealthyDestinationNamespaces"
+)
+
+// Reasons
+const (
+	ErrorsDetected = "ErrorsDetected"
+)
+
+// Messages
+const (
+	UnhealthyDestination = "The destination cluster has unhealthy migrated namespaces []. See status.namespaces for details."
+	UnhealthySource      = "The source cluster has unhealthy namespaces []. See status.namespaces for details."
+)
+
+const (
+	// ImagePullBackOff is unhealthy container state
+	ImagePullBackOff = "ImagePullBackOff"
+	// CrashLoopBackOff is unhealthy container state
+	CrashLoopBackOff = "CrashLoopBackOff"
+)
+
+// UnhealthyResources is a store for unhealthy namespaces across clusters
+type UnhealthyResources struct {
+	Namespaces []UnhealthyNamespace `json:"namespaces,omitempty"`
+}
+
+// Workload is a store for unhealthy resource and it's dependents
+type Workload struct {
+	Name      string   `json:"name"`
+	Resources []string `json:"resources,omitempty"`
+}
+
+// UnhealthyNamespace is a store for unhealthy resources in a namespace
+type UnhealthyNamespace struct {
+	Name      string     `json:"name"`
+	Workloads []Workload `json:"workloads"`
+}
+
+// IsHealthy will report if there are any unhealthy namespaces were present
+func (m *MigMigration) IsHealthy() bool {
+	return m.Status.UnhealthyResources.Namespaces == nil || len(m.Status.UnhealthyResources.Namespaces) == 0
+}
+
+// VerifyPods would collect, display and notify about the pod health in a specific namespace
+func (u *UnhealthyResources) VerifyPods(client k8sclient.Client, options *k8sclient.ListOptions) (*[]unstructured.Unstructured, error) {
```

Comment:
`VerifyPods()` seems like a method on the Reconciler not the `UnhealthyResources` object.

Reply:
So, this mean that until the `spec.verify` is set on the migPlan, the migrations will verify target namespaces on every reconcile?

Reply:
I suppose this part could be used at any moment, and not only by migration/plan reconsile.

Reply:
No, the migration controller will only do the verification when verify=true and the migration is a _final_ migration that has completed successfully.